lib/Container/Compiler/FieldTypeRegistryPass.php

```diff
@@ -25,8 +25,6 @@ public function process(ContainerBuilder $container)
         $fieldTypeRegistry = $container->findDefinition('ezpublish.persistence.field_type_registry');
         $fieldTypeRegistry->setClass($fieldTypeRegistryClass);
 
-        if (count($fieldTypeRegistry->getArguments()) > 0) {
-            $fieldTypeRegistry->setArgument(0, $fieldTypeRegistryClass);
-        }
+        $fieldTypeRegistry->replaceArgument(0, $fieldTypeRegistryClass);
```

Comment:
Not sure, but I think this will break if there's no argument at index `0` (eZ Platform v3 case), so the `count` check is still needed.

Reply:
You're right, fixed in https://github.com/netgen/ezplatform-search-extra/pull/28/commits/d73d66966a9aa6d440755a3d8a7444ea0870014f